client/src/app/shared/header/header.component.ts

```diff
@@ -1,15 +1,35 @@
 import { Component, OnInit } from '@angular/core';
+import { AppRoutes } from '../../app-routes.module';
 
 @Component({
   selector: 'app-header',
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.css']
 })
 export class HeaderComponent implements OnInit {
+  routes: Array<any> = AppRoutes;
 
   constructor() { }
 
   ngOnInit() {
+    this.routes.map((route) => {
+      let routeName = route.component.name;
+      
+      if(routeName && routeName.length > 9){
+        route.name = routeName.substr(0, routeName.length - 9);
+      } else {
+        route.name = route.path;
+      }
+
+      if(route.children){
+        route.subroutes = [];
+      
+        route.children.map((child) => {
+          let childName = child.component.name
```

Comment:
Need to validate childName length before using substr function?

Reply:
Thanks, yes. I did not setup any unit test on this one, will work to set them now.

Reply:
Also. Array.prototype.map return a new String and it's callback function required a return value. If you only want to traverse the Array, forEach would be better I think.